### PR TITLE
Feature rochester corrections

### PR DESCRIPTION
**Pull request by HeinerTholen**
Adding post-processing module for Rochester corrections for 2016 and 2017.

### CODE REVIEW

File: python/postprocessing/modules/common/muonScaleResProducer.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    def endFile(self, inputFile, outputFile, inputTree, wrappedOutputTree):
+        pass
+
+    def analyze(self, event):
+        muons = Collection(event, "Muon")
+        roccor = self._roccor
+        try:
+            is_mc = bool(getattr(event, 'GenJet_pt'))
+        except RuntimeError:
+            is_mc = False
```
Comment (gpetruc) [inline python/postprocessing/modules/common/muonScaleResProducer.py]: can the is_mc be derived once per job in beginFile, checking if inputTree.GetBranch("GenJet_pt") is there?

**Comment by arizzi** [comment]
is the .cc .h code auto-generated and run-period dependent? Or is it generic code that can be hosted outside data/ directory?

File: python/postprocessing/modules/common/muonScaleResProducer.py (hunk 1 above)
Comment (HeinerTholen) [reply python/postprocessing/modules/common/muonScaleResProducer.py]: Thanks!
(just pushed.. )

**Comment by HeinerTholen** [comment]
It's the run period dependent code provided by the Rochester guys. I put these folders here as the come and if it's not a big deal to you, I'd like to leave it together.


Am 13. April 2018 15:25:01 MESZ schrieb arizzi <notifications@github.com>:
>is the .cc .h code auto-generated and run-period dependent? Or is it
>generic code that can be hosted outside data/ directory?
>
>-- 
>You are receiving this because you authored the thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/cms-nanoAOD/nanoAOD-tools/pull/71#issuecomment-381134188

-- 
Sent from my phone. Please excuse my brevity.

**Comment by arizzi** [comment]
are the two file in 2016 and 2017 version different?

**Comment by HeinerTholen** [comment]
Yes, the 2016 version has a folder with a good number of separate text files and the 2017 version combines it into one. I asked the Rochester guys if they can provide the 2016 correction factors in the single-file-format of 2017, but haven't gotten an answer on that yet. Once it comes, I'll update this code.

**Pull request merged**